File: client_web/src/app/pages/area/area.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';

import {ReactionViewModel} from '../../viewModels/area/ReactionViewModel';
import {ResultViewModel} from '../../viewModels/ResultViewModel';
import {NewAreaViewModel} from '../../viewModels/area/NewAreaViewModel';

import { Account } from '../../models/account';
import {Trigger} from '../../models/trigger';
import {ActionViewModel} from '../../viewModels/area/ActionViewModel';

@Injectable({ providedIn: 'root' })
export class AreaService {
  constructor(private http: HttpClient) {}

  public getActions(): Promise<ActionViewModel[]> {
    return this.http.get<ActionViewModel[]>('/api/area/actions/').toPromise().then(
      (result: ActionViewModel[]) => {
        return result;
      }, (error) => {
        console.log('AreaService(getActions): Error ' + error);
        return [];
      }
    );
  }

  public getReactions(): Promise<ReactionViewModel[]> {
    return this.http.get<ReactionViewModel[]>('/api/area/reactions/').toPromise().then(
      (result: ReactionViewModel[]) => {
        return result;
      }, (error) => {
        console.log('AreaService(getReactions): Error ' + error);
        return [];
      }
    );
  }

  public newArea(account: Account, actionId: number, reactionId: number): Promise<ResultViewModel> {
    const model: NewAreaViewModel = new NewAreaViewModel();
    model.token = account.token;
    model.reactionId = reactionId;
    model.actionId = actionId;
    console.log(actionId + ' ' + reactionId);
    const body = JSON.stringify(model);
    return this.http.post('/api/area/new', body).toPromise().then(
      (result: ResultViewModel) => {
        return result;
      },
      (error) => {
        console.log('AreaService(newArea): Error ' + error);
        const result = new ResultViewModel();
        result.error = 'Une erreur sais produite';
        result.success = false;
        return result;
      }
    );
  }

  public getTriggers(account: Account): Promise<Trigger[]> {
    const body = JSON.stringify(account);
    return this.http.post('/api/area/triggers', body).toPromise().then(
      (result: Trigger[]) => {
        return result;
      },
      (error) => {
        console.log('AreaService(getTriggers): Error ' + error);
        return [];
      }
    );
  }

  public deleteTrigger(account: Account, id: number): Promise<ResultViewModel> {
    const body = JSON.stringify(account);
    return this.http.post<ResultViewModel>('/api/area/delete/trigger/' + id, body).toPromise().then(
      (result: ResultViewModel) => {
        return result;
      },
      (error) => {
        console.log('AreaService(getTriggers): Error ' + error);
        const result = new ResultViewModel();
        result.error = 'Une erreur sais produite';
        result.success = false;
        return result;
      }
    );
  }
}

File: mobile_user/app/src/main/java/area/epitech/area/Services/AreaService.kt
package area.epitech.area.Services

import android.util.Log
import area.epitech.area.ViewModels.ConnectedViewModel
import com.github.kittinunf.fuel.Fuel
import com.github.kittinunf.fuel.core.Request
import com.github.kittinunf.fuel.core.extensions.cUrlString
import com.google.gson.Gson

class AreaService {
    private val TAG = AreaService::class.java.simpleName
    companion object {
        val instance: AreaService = AreaService()
    }


    fun deleteTrigger(token: String, id: Int): Request {
        val model = ConnectedViewModel()
        model.Token = token
        Log.d(TAG, "Getting triggers request...")
        val json = Gson().toJson(model)
        Log.d(TAG, "Model Json: $json")
        return Fuel.post("/api/area/delete/trigger/" + id)
            .body(json)
            .also { Log.d(TAG, it.cUrlString()) }

    }

    fun getTriggers(token: String): Request {
        val model = ConnectedViewModel()
        model.Token = token
        Log.d(TAG, "Getting triggers request...")
        val json = Gson().toJson(model)
        Log.d(TAG, "Model Json: $json")
        return Fuel.post("/api/area/triggers")
            .body(json)
            .also { Log.d(TAG, it.cUrlString()) }

    }

    fun getActions(): Request {
        Log.d(TAG, "Getting actions request...")
        return Fuel.get("/api/area/actions")
            .also { Log.d(TAG, it.cUrlString()) }

    }

    fun getReactions(): Request {
        Log.d(TAG, "Getting reactions request...")
        return Fuel.get("/api/area/reactions")
            .also { Log.d(TAG, it.cUrlString()) }

    }
}